### PR TITLE
Issue #2742207: Port commerce_product tests to BrowserTest

### PR DESCRIPTION
**Pull request by smmccabe**
Ported and working, the one failure is a legit failure and I will create another pull request when I figur e out the fix, possible blocking but wanted to get this posted.

### CODE REVIEW

File: modules/product/commerce_product.module [hunk 1]
```diff
@@ -45,13 +45,30 @@ function commerce_product_entity_form_display_update(EntityFormDisplayInterface
...
+/**
  * Implements hook_ENTITY_TYPE_update().
  */
 function commerce_product_entity_view_display_update(Drupal\Core\Entity\EntityInterface $entity) {
   // The product view uses the variation view and needs to be cleared, which doesn't happen automatically
   // because we're editing the variation, not the product
   if (substr($entity->getConfigTarget(), 0, 27) === 'commerce_product_variation.') {
     Cache::invalidateTags(['commerce_product_view']);
+    //if (\Drupal::entityManager()->hasHandler($entity->getConfigTarget(), 'view_builder')) {
+      \Drupal::entityManager()->getViewBuilder($entity->getConfigTarget())->resetCache();
```
Comment (mglaman) [inline modules/product/commerce_product.module]: This is preventing installs

**Comment by smmccabe** [comment]
Yah, my previous cache fix that got commited works fine on form submit, but doesn't seem to work on a straight save, still attempting to get a all-cases fix working

**Comment by smmccabe** [comment]
After discussion with mglaman, just decided to change the test to submit to clear cache as neither of us could figure out a way to make it work without that.

File: commerce.info.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@ package: Commerce
 core: 8.x
 configure: commerce.admin_commerce
 dependencies:
-  - entity
+  - address
+  - entity:entity
+  - inline_entity_form
```
Comment (mglaman) [inline commerce.info.yml:10]: Wait a minute

File: composer.json [hunk 3]
```diff
@@ -12,7 +12,6 @@
         "drupal/profile": "8.1.x-dev",
         "drupal/state_machine": "8.1.x-dev",
         "commerceguys/intl": "dev-master",
-        "commerceguys/pricing": "dev-master",
```
Comment (mglaman) [inline composer.json]: Wait a minute

File: modules/order/commerce_order.info.yml [hunk 4]
```diff
@@ -8,7 +8,6 @@ dependencies:
   - commerce_price
   - commerce_store
   - datetime
-  - inline_entity_form
```
Comment (mglaman) [inline modules/order/commerce_order.info.yml]: Wait a minute

**Pull request merged**